static/app/components/events/autofix/autofixInsightCards.tsx

```diff
@@ -260,46 +238,142 @@ interface AutofixInsightCardsProps {
   insights: AutofixInsight[];
   runId: string;
   stepIndex: number;
-  shouldCollapseByDefault?: boolean;
 }
 
-function CollapsibleChainLink({
-  isEmpty,
-  isCollapsed,
-  onToggleCollapse,
-  insightCount,
-}: {
+interface CollapsibleChainLinkProps {
+  groupId: string;
+  runId: string;
+  stepIndex: number;
+  alignment?: 'start' | 'center';
   insightCount?: number;
   isCollapsed?: boolean;
   isEmpty?: boolean;
   onToggleCollapse?: () => void;
-}) {
+  showAddControl?: boolean;
+  showCollapseControl?: boolean;
+}
+
+function CollapsibleChainLink({
+  insightCount,
+  isCollapsed,
+  isEmpty,
+  onToggleCollapse,
+  showAddControl,
+  showCollapseControl,
+  stepIndex,
+  groupId,
+  runId,
+  alignment = 'center',
+}: CollapsibleChainLinkProps) {
+  const [isAdding, setIsAdding] = useState(false);
+  const [newInsightText, setNewInsightText] = useState('');
+  const {mutate: updateInsight} = useUpdateInsightCard({groupId, runId});
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    setIsAdding(false);
+    updateInsight({
+      message: newInsightText,
+      step_index: stepIndex,
+      retain_insight_card_index:
+        insightCount !== undefined && insightCount > 0 ? insightCount - 1 : null,
+    });
+    setNewInsightText('');
+  };
+
+  const handleCancel = () => {
+    setIsAdding(false);
+    setNewInsightText('');
+  };
+
   return (
-    <VerticalLineContainer isEmpty={isEmpty}>
-      <VerticalLine />
-      <RethinkButtonContainer className="rethink-button-container">
-        {onToggleCollapse && (
-          <CollapseButtonWrapper>
-            {isCollapsed && insightCount && insightCount > 0 && (
+    <VerticalLineContainer isEmpty={isEmpty} alignment={alignment}>
+      <RethinkButtonContainer
+        className="rethink-button-container"
+        parentAlignment={alignment}
+      >
+        {showCollapseControl && onToggleCollapse && (
+          <CollapseButtonWrapper
+            onClick={onToggleCollapse}
+            title={isCollapsed ? t('Show reasoning') : t('Hide reasoning')}
+            aria-label={t('Toggle reasoning visibility icon')}
+          >
+            {isCollapsed && insightCount && insightCount > 0 ? (
               <CollapsedCount>
                 {tn('%s insight hidden', '%s insights hidden', insightCount)}
               </CollapsedCount>
+            ) : (
+              <CollapsedCount>{}</CollapsedCount>
```

Comment:
I'm assuming the reason for this is because this is expecting a child?

Reply:
yep